src/test/java/no/unit/nva/cristin/projects/FetchCristinProjectsTest.java

```diff
@@ -450,6 +472,62 @@ void handlerReturnsProjectsWrapperWithCorrectNextResultsAndPreviousResultsWhenLi
         assertEquals(expectedPrevious, actualPrevious);
     }
 
+    @Test
+    void handlerReturnsMatchingProjectsFromGrantIdSearchWhenSuppliedWithOnlyNumber() throws Exception {
```

Comment:
I am not sure what we are testing here. I changed the value "GRANT_ID_EXAMPLE" with a random string ("sdfsf") and the test passed

Reply:
I think the testing has become very complicated because we are testing both the client and the mocking server. 

I think the idea is that "if the query is a number search first for ids and if empty search for title", so basically we should set up the mock server to return empty in the first call and something in the second.

Reply:
See other test named: `handlerReturnsProjectsFromTitleSearchWhenGrantIdSearchReturnsZeroResults`

Reply:
It was a bug. It was to throw an exception if it does a query on title because it should only do query on grant id and not title. It should work now after latest commit